hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileGetContentsHandler.java

```diff
@@ -16,54 +16,99 @@
  */
 @Singleton
-public class FileGetContentsHandler extends PaidQueryHandler {
+public class FileGetContentsHandler extends FileQueryBase {
+    private final NetworkInfo networkInfo;
+
     @Inject
-    public FileGetContentsHandler() {
-        // Exists for injection
+    public FileGetContentsHandler(@NonNull final NetworkInfo networkInfo) {
+        this.networkInfo = requireNonNull(networkInfo);
     }
 
     @Override
-    public QueryHeader extractHeader(@NonNull final Query query) {
+    public @NonNull QueryHeader extractHeader(@NonNull final Query query) {
         requireNonNull(query);
         return query.fileGetContentsOrThrow().header();
     }
 
     @Override
-    public Response createEmptyResponse(@NonNull final ResponseHeader header) {
+    public @NonNull Response createEmptyResponse(@NonNull final ResponseHeader header) {
         requireNonNull(header);
         final var response = FileGetContentsResponse.newBuilder().header(header);
         return Response.newBuilder().fileGetContents(response).build();
     }
 
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
-        requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final FileGetContentsQuery op = query.fileGetContentsOrThrow();
+        if (op.hasFileID()) {
+            validateFileExistence(op.fileID(), context);
+        }
     }
 
     @Override
-    public Response findResponse(@NonNull final QueryContext context, @NonNull final ResponseHeader header) {
-        requireNonNull(context);
+    public @NonNull Response findResponse(@NonNull final QueryContext context, @NonNull final ResponseHeader header) {
         requireNonNull(header);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var fileStore = context.createStore(ReadableFileStoreImpl.class);
+        final var op = query.fileGetContentsOrThrow();
+        final var responseBuilder = FileGetContentsResponse.newBuilder();
+        final var file = op.fileIDOrElse(FileID.DEFAULT);
+
+        final var responseType = op.headerOrElse(QueryHeader.DEFAULT).responseType();
+        responseBuilder.header(header);
+        if (header.nodeTransactionPrecheckCode() == OK && responseType != COST_ANSWER) {
+            final var optionalInfo = contentFile(file, fileStore);
+            optionalInfo.ifPresent(responseBuilder::fileContents);
+        }
+
+        return Response.newBuilder().fileGetContents(responseBuilder).build();
+    }
+
+    /**
+     * Provides file content about a file.
+     * @param fileID the file to get information about
+     * @param fileStore the file store
+     * @return the content about the file
+     */
+    private @Nullable Optional<FileContents> contentFile(
```

Comment:
could be static. Maybe it makes sense to provide a util class with some static methods

Reply:
but it is just used meanwhile in quey get content info. if it will be used somewhere else you are definitely right